piptools/scripts/compile.py

```diff
@@ -348,7 +348,7 @@ def cli(
         resolver = Resolver(
             constraints,
             repository,
-            prereleases=pre,
+            prereleases=pre or repository.finder.allow_all_prereleases or None,
```

Comment:
I think i would be better in this way:
```suggestion
            prereleases=repository.finder.allow_all_prereleases or pre,
```